lib/iris/tests/unit/test_sample_data_path.py

```diff
@@ -61,57 +60,42 @@ def test_call(self):
-
-    def tearDown(self):
-        shutil.rmtree(self.sample_dir)
-
-    def test_file_ok(self):
-        sample_path = _temp_file(self.sample_dir)
-        result = sample_data_path(os.path.basename(sample_path))
-        self.assertEqual(result, sample_path)
-
     def test_file_not_found(self):
-        with self.assertRaisesRegexp(ValueError, 'Sample data .* not found'):
-            sample_data_path('foo')
+        with mock.patch('iris_sample_data.path', self.sample_dir):
+            with self.assertRaisesRegexp(ValueError,
+                                         'Sample data .* not found'):
+                sample_data_path('foo')
 
     def test_file_absolute(self):
-        with self.assertRaisesRegexp(ValueError, 'Absolute path'):
-            sample_data_path(os.path.abspath('foo'))
+        with mock.patch('iris_sample_data.path', self.sample_dir):
+            with self.assertRaisesRegexp(ValueError, 'Absolute path'):
+                sample_data_path(os.path.abspath('foo'))
 
     def test_glob_ok(self):
         sample_path = _temp_file(self.sample_dir)
         sample_glob = '?' + os.path.basename(sample_path)[1:]
-        result = sample_data_path(sample_glob)
-        self.assertEqual(result, os.path.join(self.sample_dir, sample_glob))
+        with mock.patch('iris_sample_data.path', self.sample_dir):
+            result = sample_data_path(sample_glob)
+            self.assertEqual(result, os.path.join(self.sample_dir,
+                                                  sample_glob))
 
     def test_glob_not_found(self):
-        with self.assertRaisesRegexp(ValueError, 'Sample data .* not found'):
-            sample_data_path('foo.*')
+        with mock.patch('iris_sample_data.path', self.sample_dir):
+            with self.assertRaisesRegexp(ValueError,
+                                         'Sample data .* not found'):
+                sample_data_path('foo.*')
 
     def test_glob_absolute(self):
-        with self.assertRaisesRegexp(ValueError, 'Absolute path'):
-            sample_data_path(os.path.abspath('foo.*'))
+        with mock.patch('iris_sample_data.path', self.sample_dir):
+            with self.assertRaisesRegexp(ValueError, 'Absolute path'):
+                sample_data_path(os.path.abspath('foo.*'))
 
-    def test_warn_deprecated(self):
-        sample_path = _temp_file(self.sample_dir)
-        with mock.patch('warnings.warn') as warn:
-            sample_data_path(os.path.basename(sample_path))
-            self.assertEqual(warn.call_count, 1)
-            (warn_msg, warn_exception), _ = warn.call_args
-            msg = 'iris.config.SAMPLE_DATA_DIR was deprecated'
-            self.assertTrue(warn_msg.startswith(msg))
-            self.assertEqual(warn_exception, IrisDeprecation)
+
+class TestIrisSampleDataMissing(tests.IrisTest):
+    def test_no_iris_sample_data(self):
+        self.patch('iris.iris_sample_data', None)
+        with self.assertRaisesRegexp(ValueError, 'Please install'):
```

Comment:
If you choose to change the error class above you'll need to change it here too.